src/main/groovy/com/github/spotbugs/snom/SpotBugsExtension.groovy

```diff
@@ -98,7 +99,8 @@ class SpotBugsExtension {
      * <p>See also <a href="https://spotbugs.readthedocs.io/en/stable/filter.html">SpotBugs Manual about Filter file</a>.</p>
      */
     @NonNull
-    final RegularFileProperty includeFilter;
+    final Property<Object> includeFilter;
```

Comment:
this field is public API, so it's better to keep it compatible. I suggest you to add a new field, and link `includeFilter` prop to it via a convention. like:

```groovy
includeFilter.convention(() -> downloadFrom(includeFilterUrl))
```

or, it's better to break backward compatibility, make next release as a major update (v5.0.0). Refer https://www.conventionalcommits.org/en/v1.0.0/ to learn how to release `BREAKING CHANGE`.

Reply:
I don't understand your comment by keeping it compatible. The property is backwards compatible the way I wrote the code. All you loose is strongly typed RegularFileProperty, but I didn't see any other way for this one setting to accept to different types.

If you add new property, includeFilterUrl, then you will also need extra user input validation to make sure both things are not specified at the same time. Also I don't really understand that "convention()" thing and Gradle has rather poor documentation.